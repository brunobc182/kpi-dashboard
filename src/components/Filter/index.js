import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const Wrapper = styled.div`
  padding: 6px 0;
  display: flex;
  align-items: center;
  justify-content: space-around;
  width: 100%;
  max-width: 300px;
`;

const Label = styled.label`
  margin-right: 12px;
  width: 40%;
`;

const Select = styled.select`
  padding: 6px 12px;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 100%;
  max-width: 230px;
  min-width: 190px;
  border-radius: 4px;
  text-align-last: center;
`;

const Option = styled.option``;

function Filter({ options, testeId, selectectValue, handleChange }) {
  return (
    <Wrapper>
      {options.length !== 0 ? (
        <>
          <Label>Filter by</Label>
          <Select
            data-testid={testeId}
            value={selectectValue}
            onChange={handleChange}
          >
            {options.map((option) => (
              <Option key={option.value} value={option.value}>
                {option.text}
              </Option>
            ))}
          </Select>
        </>
      ) : (
        "There is no option to filter"
      )}
    </Wrapper>
  );
}

Filter.propTypes = {
  options: PropTypes.arrayOf(
    PropTypes.shape({
      time_unit: PropTypes.string,
      time_unit_count: PropTypes.number,
    })
  ),
  handleChange: PropTypes.func,
  selectectValue: PropTypes.any,
  testId: PropTypes.string,
};

Filter.defaultProps = {
  options: [],
  handleChange: () => {},
  selectectValue: "selectedValue",
  testId: "filterId",
};

export default Filter;
