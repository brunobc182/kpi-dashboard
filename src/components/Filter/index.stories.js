import React from "react";
import Filter from "./";

const selectOptions = [
  {
    text: "WEEK",
    value: 1,
  },
  {
    text: "MONTH",
    value: 2,
  },
  {
    text: "QUARTER",
    value: 3,
  },
  {
    text: "YEAR",
    value: 4,
  },
];

export const WithoutProps = () => <Filter />;
export const WithProps = () => <Filter options={selectOptions} />;

export default {
  title: "Components/Filter",
  component: Filter,
};
