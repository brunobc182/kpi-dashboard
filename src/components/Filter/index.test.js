import React from "react";
import { render } from "@testing-library/react";
import Filter from "./";

describe("Filter", () => {
  it("renders correctly", () => {
    const selectOptions = [
      {
        time_unit: "WEEK",
        time_unit_count: 1,
      },
      {
        time_unit: "MONTH",
        time_unit_count: 1,
      },
      {
        time_unit: "QUARTER",
        time_unit_count: 1,
      },
      {
        time_unit: "YEAR",
        time_unit_count: 1,
      },
    ];
    const { container } = render(<Filter options={selectOptions} />);

    expect(container).toMatchSnapshot();
  });

  it("renders correctly with no options supplied", () => {
    const { container, getByText } = render(<Filter />);
    const empytFilterListMsg = getByText(/There is no option to filter/i);

    expect(empytFilterListMsg).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
