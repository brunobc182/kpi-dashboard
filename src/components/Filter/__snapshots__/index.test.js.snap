// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
<div>
  <div
    class="sc-AxjAm iEFAbG"
  >
    <label
      class="sc-AxirZ hriSXr"
    >
      Filter by
    </label>
    <select
      class="sc-AxiKw bJukMX"
    >
      <option
        class="sc-AxhCb ccMvAG"
      />
      <option
        class="sc-AxhCb ccMvAG"
      />
      <option
        class="sc-AxhCb ccMvAG"
      />
      <option
        class="sc-AxhCb ccMvAG"
      />
    </select>
  </div>
</div>
`;

exports[`Filter renders correctly with no options supplied 1`] = `
<div>
  <div
    class="sc-AxjAm iEFAbG"
  >
    There is no option to filter
  </div>
</div>
`;
